servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/PipelinedStreamingHttpConnection.java

```diff
@@ -19,38 +19,35 @@
 import io.servicetalk.concurrent.api.Publisher;
 import io.servicetalk.http.api.HttpExecutionContext;
 import io.servicetalk.http.api.StreamingHttpRequestResponseFactory;
-import io.servicetalk.transport.netty.internal.DefaultNettyPipelinedConnection;
 import io.servicetalk.transport.netty.internal.FlushStrategy;
 import io.servicetalk.transport.netty.internal.NettyConnection;
+import io.servicetalk.transport.netty.internal.WriteDemandEstimators;
 
 import javax.annotation.Nullable;
 
 final class PipelinedStreamingHttpConnection
-        extends AbstractStreamingHttpConnection<DefaultNettyPipelinedConnection<Object, Object>> {
-
-    private final NettyConnection<Object, Object> nettyConnection;
-
+        extends AbstractStreamingHttpConnection<NettyPipelinedConnection<Object, Object>> {
     PipelinedStreamingHttpConnection(final NettyConnection<Object, Object> connection,
                                      final H1ProtocolConfig config,
                                      final HttpExecutionContext executionContext,
                                      final StreamingHttpRequestResponseFactory reqRespFactory) {
-        super(new DefaultNettyPipelinedConnection<>(connection, config.maxPipelinedRequests()),
+        super(new NettyPipelinedConnection<>(connection),
                 config.maxPipelinedRequests(), executionContext, reqRespFactory, config.headersFactory());
-        this.nettyConnection = connection;
     }
 
     @Override
     protected Publisher<Object> writeAndRead(Publisher<Object> requestStream,
                                              @Nullable final FlushStrategy flushStrategy) {
         if (flushStrategy == null) {
-            return connection.request(requestStream);
+            return connection.write(requestStream, connection::defaultFlushStrategy,
```

Comment:
I wanted to keep the new API exposure for the `NettyPipelinedConnection` low for now, but I think this is worth revisiting as part of the larger "how is state propagated on the server https://github.com/apple/servicetalk/issues/981" which may provide opportunities to clean this up.